gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -16,6 +16,10 @@
 
 package com.google.cloud.hadoop.gcsio;
 
+import static com.google.cloud.hadoop.gcsio.CloudMonitoringMetricsRecorder.LATENCY_MS;
+import static com.google.cloud.hadoop.gcsio.CloudMonitoringMetricsRecorder.PROTOCOL_JSON;
+import static com.google.cloud.hadoop.gcsio.CloudMonitoringMetricsRecorder.recordErrorMetric;
+import static com.google.cloud.hadoop.gcsio.CloudMonitoringMetricsRecorder.recordSuccessMetric;
```

Comment:
Why is this static method? I think it should be instance method on of the `CloudMonitoringMetricsRecorder` instance.

Reply:
For instance method, we would need to move it to `MetricsRecorder` interface. Success and error metrics seemed too specific to add instance methods. Since their functionalities were on top of actual metric interfaces, I extracted them as util methods to `CloudMonitoringMetricsRecorder` (for lack of a better place)

We could still expose these methods as instance methods from `MetricsRecorder`, would you prefer that ?

Reply:
If they are too specific for all types of metric syncs/recorders, then maybe we abstracted interface too soon? And should stick just with an impl class for just Cloud Monitoring sync first and extract common interface when we add more syncs and it will be more clear how common interface could look like? Also, I think that this defeats point of having `MetricsRecorder` interface, as we will need to wire similar static helper methods for other metric syncs in the future, while we should strive to opposite - ability to implement new syncs without need to do any changes at the call sites of `MetricsRecorder` interface and just add new implementation of the `MetricsRecorder` interface.

Alternative will be to create some wrapper class that takes `CloudMonitoringMetricsRecorder` instance as a constructor parameter so we can have instance helper methods for it that do not require passing `CloudMonitoringMetricsRecorder` instance for each method call - this seems to be more OOP-like, but still not as good as having these methods in the `CloudMonitoringMetricsRecorder` directly, if possible.